_posts/2015-08-24-Cocoa-Architecture:-Hybrid-Apps.markdown

```diff
@@ -0,0 +1,68 @@
+---
+layout: post
+title: 'Cocoa Architecture: Hybrid Apps'
+date: 2015-08-24T00:00:00.000Z
+comments: false
+categories:
+  - ios
+  - mobile
+  - architecture
+  - eigen
+  - hybrid
+author: Orta Therox
+github-url: 'https://www.github.com/orta'
+twitter-url: 'http://twitter.com/orta'
+blog-url: 'http://orta.io'
+---
+
+As a part of going through the design patterns we've found in the creation of the Artsy iOS apps, I'd like to talk a bit about Hybrid Applications. A hybrid application refers to an app that uses native code and web content intertwined. Our flagship iOS app, [eigen](https://github.com/artsy/eigen) is a hybrid app, and it seems to get more and more hybrid-y each release. Let's talk a little bit about the pros and cons of this. <!-- more -->
+
+----
+
+### What is a Hybrid App
+
+I gave the widest possible definition above, so let's dig in as this can be a contentious. There is a whole spectrum of which an app can be classed as a hybrid app. This ranges from; more or less web-tech everywhere to 100% Objective-C.
+
+A great example of the furthest to the web-side is [ATOM](https://atom.io), the text editor. It's on the extreme side because all of the user interface is built using HTML + CSS, and almost all of the app is in javascript. The trade-off for them is that they can easy write cross-platform code, that will work reliably with technology that the vast majority of programmers use.
```

Comment:
The trade-off could be expanded – without any downsides mentioned, this makes me think it's a great idea :)